python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -963,12 +961,6 @@ def _standardizeIngestPath(
         if not srcUri.isabs():
             srcUri = self.root.join(path)
 
-        if not srcUri.exists():
-            raise FileNotFoundError(
-                f"Resource at {srcUri} does not exist; note that paths to ingest "
-                f"are assumed to be relative to {self.root} unless they are absolute."
```

Comment:
I can't see anyway around removing this. Maybe I should do the check if the ingest is direct or None and then only if local file? There is no reason to do it for a copy since the copy will fail later on anyhow.

Reply:
This doesn't seem like a primary function of this method, given its name and description.  I'm fine with a check (if any) happening elsewhere.

Reply:
It sounds like any check is going to be prohibitively slow. Options seem to be:

* Never check and hope the file really is there when you ingest with direct / no-track-file-attrs
* Always check if local else never check
* Only check if there are fewer than 200 files being ingested
* Always check the first 10 (for example) regardless and if those do exist trust that the person has a reasonable set of data that is likely to really exist.

I think I'm talking myself into the last option (maybe above this method). That would catch the egregious case where someone is trying to ingest without really having access to the remote bucket or they copied the export file to their laptop and are assuming the files are still accessible. Even only checking the first might be enough.

cc/ @TallJimbo in case you have an opinion.

Reply:
This is all for `direct` and `None` transfer modes, and not the others, right?  Not some case where a genuine artifact transfer silently fails?  If that's the case then some sort of spot-checking like the last bullet sounds reasonable.  Maybe the first one and a random 1% or something thereafter?

Reply:
Right. If "copy" mode is being used then the ingest will break as soon as it tries to copy the file (maybe "symlink" will also fail?). If we are tracking file attributes the ingest will fail when it can't find the size.

So we only really need to do any spot checks if we are using direct/None with no-track-file-attrs.

Reply:
Ok. I've changed it to only check 200 of the files being ingested. That doesn't take very long and you can at least get a sense that things are probably okay.